### PR TITLE
Reacting to Hosting changes

### PR DESCRIPTION
**Pull request by JunTaoLuo**
reacting to https://github.com/aspnet/Hosting/pull/445

### CODE REVIEW

File: samples/Owin.Nowin.HelloWorld/NowinServer.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+namespace NowinWebSockets
+{
+    public class NowinServer : IServer
+    {
+        private RequestDelegate _callback;
+        private INowinServer _nowinServer;
+
+        public IFeatureCollection Features { get; }
+
+        private Task HandleRequest(IDictionary<string, object> env)
```
Comment (davidfowl) [inline samples/Owin.Nowin.HelloWorld/NowinServer.cs]: Move this to the bottom of the file.

File: samples/Owin.Nowin.HelloWorld/NowinServer.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        public NowinServer(IFeatureCollection features)
+        {
+            Features = features;
+        }
+
+        public void Start(RequestDelegate requestDelegate)
+        {
+            var builder = ServerBuilder.New()
```
Comment (davidfowl) [inline samples/Owin.Nowin.HelloWorld/NowinServer.cs]: I would actually split this up. Do this configuration in CreateServer. The only job of IServer would be to call SetOwinApp and Start on the builder

**Comment by JunTaoLuo** [comment]
updated

**Comment by JunTaoLuo** [comment]
cc @lodejard

**Comment by JunTaoLuo** [comment]
cc @Tratcher

**Comment by JunTaoLuo** [comment]
@davidfowl gave me a :shipit: in person so I'll merge this.

**Pull request merged**